### DISCUSSION TITLE
Question :f# casting to anonymous record from object

### DISCUSSION THREAD

**Discussion by Xyncgas** [fsharp/fslang-suggestions#1286] [General]
If I have {|A:TA;B:TB} as an ```obj```
and I am casting this .NET object to {|B:TB;A:TA;|}, can I expect overheads from translating between different layout or is it smart enough to just give me the original thing

**Comment by Happypig375** [comment]
Anonymous records are structural. Those two types are the same type.

**Comment by SchlenkR** [comment]
> Anonymous records are structural

May I add: As far as I know, they are nominal, but inferred to be the same nominal type based on their structure, independent of the order of properties when creating an instance (at least in the same assembly). 

Since they are nominal, assembly boundaries could play a role here, which could mean that they might not(!) be inferred to be the same nominal type from different assemblies.

**Reply by baronfel** [reply]
@Xyncgas if you are curious, remember you can always use sharplab to answer these questions yourself. Here's your question in sharplab, for example: https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AbEAzAzgHwgAcYA7AAgGUBPXAFxgFsBYAKDYxjvLHIF5yAb3zkAgv3IAiABYBLSQG5yAIQl0oAVxjl8AXw5ceAfQjAAVhOAQEPNm0YBDOmGnHTFgO6y60tiJAA/EIioiDk9FCypADmSsphVhAYOrrkDrjkGqRWCDAAJuRwAHzkABQAlH7kRoUlFWxAA===

You can see that the anon type is cast/checked as the same nominal type, like @RonaldSchlenker mentioned. That means very little, if any, overhead for unboxing.

**Reply by smoothdeveloper** [reply]
note: I believe there is also a caveat that they are structural within a given assembly but not across boundaries.

**Reply by Xyncgas** [reply]
Thanks for the very very interesting information, god bless you
